apps/opik-backend/src/main/java/com/comet/opik/api/error/JsonInvalidFormatExceptionMapper.java

```diff
@@ -0,0 +1,16 @@
+package com.comet.opik.api.error;
+
+import com.fasterxml.jackson.databind.exc.InvalidFormatException;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.ExceptionMapper;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class JsonInvalidFormatExceptionMapper implements ExceptionMapper<InvalidFormatException> {
+
+    @Override
+    public Response toResponse(InvalidFormatException exception) {
+        log.warn("Deserialization exception: {}", exception.getMessage());
+        return Response.status(Response.Status.BAD_REQUEST).entity(exception.getMessage()).build();
```

Comment:
We must carefully to avoid a few details related to the framework or infrastructure. I believe the best would be to create a dedicated serializer for your use case.

Reply:
This error message in response has nothing to do with infrastructure. And regarding the framework, it's an open source anyway, so I'm not sure why we can't send a detailed message.

Dedicated serializer will not help here. Main pain point is debugging, now returned error swallows all details and not informative. It's just `Unable to process JSON`. Mistake could be made in any field, we can't predict all possible scenarios and prepare serializers for it.

Reply:
We just need to be careful with our exception messages in order to avoid issues here. Otherwise, we're mostly fine.

Reply:
Additinally, I recommend you using this`io.dropwizard.jersey.errors.ErrorMessage` class instead of String as entity. That's the standard error format, otherwise, our endpoints won't be consistent.

Reply:
Yeah, the point is that the error maybe too technical and not helpful at all

Reply:
In that case, maybe Boris can adjust this to a custom message, which is more informative. But he should at least parse the exception message to know which field is the problematic one. It might be tricker than just dumping the whole message.